stac_fastapi/core/stac_fastapi/core/core.py

```diff
@@ -347,20 +350,27 @@ async def all_catalogs(
                 conformance_classes=self.conformance_classes(),
             )
 
-            for catalog in temp_catalogs:
+            for i, (catalog, hit_token) in enumerate(zip(temp_catalogs, hit_tokens)):
                 # Get access control array for each catalog
                 access_control = catalog["access_control"]
                 catalog.pop("access_control")
                 # Add catalog to list if user has access
                 if int(access_control[-1]) or int(access_control[user_index]):
                     catalogs.append(catalog)
+                    print("catalogs length is ", len(catalogs))
+                    print("limit is ", limit)
                     if len(catalogs) >= limit:
-                        break
+                        print("here")
+                        print(i)
+                        if i < len(temp_catalogs) - 1:
+                            print("AND FINALLY HERE")
```

Comment:
These print statements seem to be left over from debugging and can be removed now.

Reply:
Good spot, removed unnecessary print statements